cpu/nrf5x_common/radio/nrfmin/nrfmin_gnrc.c

```diff
@@ -0,0 +1,201 @@
+
+    /* build the nrfmin header from the generic netif header */
+    res = hdr_netif_to_nrfmin(&nrfmin_hdr, pkt);
+    if (res < 0) {
+        DEBUG("[nrfmin_gnrc] send: failed to build nrfmin header\n");
+        gnrc_pktbuf_release(pkt);
+        return res;
+    }
+
+    /* create iovec of data */
+    vec_snip = gnrc_pktbuf_get_iovec(pkt, &vec_len);
+    if (vec_snip == NULL) {
+        DEBUG("[nrfmin_gnrc] send: failed to create IO vector\n");
+        gnrc_pktbuf_release(pkt);
+        return -ENOBUFS;
+    }
+
+    /* link first entry of the vector to the nrfmin header */
+    vec = (struct iovec *)vec_snip->data;
+    vec[0].iov_base = &nrfmin_hdr;
+    vec[0].iov_len = NRFMIN_HDR_LEN;
+
+    /* and finally send out the data and release the packet */
+    res = dev->dev->driver->send(dev->dev, vec, vec_len);
+    gnrc_pktbuf_release(vec_snip);
+
+    return res;
+}
+
+static gnrc_pktsnip_t *nrfmin_gnrc_recv(gnrc_netdev2_t *dev)
+{
+    int pktsize;
+    nrfmin_hdr_t *nrfmin;
+    gnrc_netif_hdr_t *netif;
+    gnrc_pktsnip_t *pkt_snip;
+    gnrc_pktsnip_t *hdr_snip;
+    gnrc_pktsnip_t *netif_snip;
+
+    /* get the size of the new packet */
+    pktsize = nrfmin_dev.driver->recv(NULL, NULL, 0, NULL);
+    if (pktsize <= 0) {
+        DEBUG("[nrfmin_gnrc] recv: error: tried to read empty packet\n");
+        return NULL;
+    }
+
+    /* allocate space in the packet buffer */
+    pkt_snip = gnrc_pktbuf_add(NULL, NULL, pktsize, GNRC_NETTYPE_UNDEF);
+    if (pkt_snip == NULL) {
+        DEBUG("[nrfmin_gnrc] recv: unable to allocate pktsnip\n");
+        return NULL;
+    }
+
+    /* read the incoming data into the packet buffer */
+    nrfmin_dev.driver->recv(NULL, pkt_snip->data, pktsize, NULL);
+
+    /* now we mark the nrfmin header */
+    hdr_snip = gnrc_pktbuf_mark(pkt_snip, NRFMIN_HDR_LEN, GNRC_NETTYPE_UNDEF);
+    if (hdr_snip == NULL) {
+        DEBUG("[nrfmin_gnrc] recv: unable to mark the nrfmin header\n");
+        gnrc_pktbuf_release(pkt_snip);
+        return NULL;
+    }
+
+    /* allocate the generic netif header and populate it with data from the
+       nrfmin header */
+    nrfmin = (nrfmin_hdr_t *)hdr_snip->data;
+    netif_snip = gnrc_netif_hdr_build((uint8_t *)&nrfmin->src_addr, 2,
+                                      (uint8_t *)&nrfmin->dst_addr, 2);
+    if (netif_snip == NULL) {
+        DEBUG("[nrfmin_gnrc] recv: unable to allocate netif header\n");
+        gnrc_pktbuf_release(pkt_snip);
+        return NULL;
+    }
+
+    netif = (gnrc_netif_hdr_t *)netif_snip->data;
+    if (nrfmin->dst_addr == NRFMIN_ADDR_BCAST) {
+        netif->flags |= GNRC_NETIF_HDR_FLAGS_BROADCAST;
+    }
+    netif->lqi = 0;
+    netif->rssi = 0;
+    netif->if_pid = plug.pid;
+    pkt_snip->type = nrfmin->proto;
+
+    /* finally: remove the nrfmin header and append the netif header */
+    gnrc_pktbuf_remove_snip(pkt_snip, hdr_snip);
+    LL_APPEND(pkt_snip, netif_snip);
+
+    return pkt_snip;
+}
+
+void nrfmin_gnrc_init(void)
```

Comment:
This applies to the whole PR: Generally, `GNRC`-related modules/folders/functions are **pre**fixed with 'gnrc_'. Did you change the order on purpose? For me, this form was less intuitive.